subprojects/core/src/test/groovy/org/gradle/internal/cleanup/BuildOutputCleanupCacheTest.groovy

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.cleanup
+
+import org.gradle.cache.internal.CacheFactory
+import org.gradle.cache.internal.DefaultCacheRepository
+import org.gradle.cache.internal.DefaultCacheScopeMapping
+import org.gradle.internal.service.DefaultServiceRegistry
+import org.gradle.internal.service.ServiceRegistryBuilder
+import org.gradle.internal.service.scopes.GlobalScopeServices
+import org.gradle.test.fixtures.file.TestNameTestDirectoryProvider
+import org.gradle.testfixtures.internal.NativeServicesTestFixture
+import org.gradle.util.GradleVersion
+import org.junit.Rule
+import spock.lang.Specification
+
+class BuildOutputCleanupCacheTest extends Specification {
```

Comment:
I don't think so. We don't annotation `AbstractWorkerProcessIntegrationSpec` and `DefaultGeneratedGradleJarCacheIntegrationTest` either even though these tests do something similar. The test also passed all coverage test. Why do you think it is needed?